src/scripting_api/util.js

```diff
@@ -0,0 +1,30 @@
+/* Copyright 2020 Mozilla Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { PDFObject } from "./pdf_object.js";
+
+class Util extends PDFObject {
+  constructor(data) {
+    super(data);
+
+    this._crackURL = data.crackURL;
+  }
+
+  crackURL(cURL) {
```

Comment:
Is this used anywhere? I can't find a caller.

Reply:
https://www.adobe.com/content/dam/acom/en/devnet/acrobat/pdfs/js_api_reference.pdf#page=716
Since I don't want to reinvent the wheel here, I'm using a wrapper around `window.URL` to do the job (see PdfSandbox.jsm at line 97 in m-c patch).